python/rmm/_lib/memory_resource.pyx

```diff
@@ -0,0 +1,292 @@
+import cython
+import os
+
+from libcpp.memory cimport unique_ptr, make_unique, shared_ptr, make_shared
+from libcpp.string cimport string
+
+
+cdef class MemoryResource:
+    cdef shared_ptr[device_memory_resource] c_obj
```

Comment:
Should this be moved to a pxd file so others can cimport it?

Reply:
Yup - done